### PR TITLE
Arc - Default scope for additional bean defining annotations is no lo…

### LINKED ISSUE #2905: Reimplement BeanDefiningAnnotation default scope
A bean defining annotation produced by `BeanDefiningAnnotationBuildItem` with a default scope value set is currently recognized as a stereotype. For example:

```java
@Path("foo")
class MyResource {
}
```
`@Path` is by default recognized as a stereotype with `@Singleton`. This works fine even if a user adds a custom scope:

```java
@RequestScoped
@Path("foo") // -> stereotype is ignored
class MyResource {
}
```
However, the app would fail if a user adds a strereotype with a different scope:

```java
@Model // -> stereotype - declares @RequestScoped
@Path("foo") // -> stereotype - declares @Singleton
class MyResource {
}
```
We could try to add the scope annotation directly if the bean class does not declare any scope annotation. But the problem is that we can't register an `AnnotationTransformer` because the custom scopes are not known during registration and so we may end up with unpredictable results. I'm not sure what the right approach would be.

### PR DESCRIPTION
**Pull request by manovotn**
…nger contesting with user defined stereotypes.

Fixes #2905 

I tested this using altered quickstart sample.

### CODE REVIEW

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java [hunk 1]
```diff
@@ -323,9 +323,15 @@ private static ScopeInfo initStereotypeScope(List<StereotypeInfo> stereotypes, A
         if (stereotypes.isEmpty()) {
             return null;
         }
-        final Set<ScopeInfo> stereotypeScopes = new HashSet<>();
-        for (StereotypeInfo stereotype : stereotypes) {
-            stereotypeScopes.add(stereotype.getDefaultScope());
+        // first go through all standard stereotypes
+        Set<ScopeInfo> stereotypeScopes = stereotypes.stream().filter(s -> !s.isAdditionalBeanDefiningAnnotation())
```
Comment (mkouba) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java]: What's wrong with adding one `if` statement to the original for loop? I mean the stream and two more lambdas don't bring any benefit here, except that it's harder to read and few more objects are created ;-).

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java [hunk 2]
```diff
@@ -323,9 +323,15 @@ private static ScopeInfo initStereotypeScope(List<StereotypeInfo> stereotypes, A
...
+                .map(stereotypeInfo -> stereotypeInfo.getDefaultScope())
+                .collect(Collectors.toSet());
+        // if the set is empty, try adding all stereotypes originating from additional bean def. ann.
+        if (stereotypeScopes.isEmpty()) {
+            stereotypeScopes.addAll(stereotypes.stream().filter(s -> s.isAdditionalBeanDefiningAnnotation())
```
Comment (mkouba) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java]: There's no need to `filter` the stream here, right?

Comment (manovotn) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java]: You either filter it or you attempt to re-add those that you already went over, which is harmless anyway.

Comment (mkouba) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java]: If it's empty then there are no non-bean defining annotations stereotypes, or?

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java (hunk 1 above)
Comment (manovotn) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java]: @mkouba better? :)

**Review by mkouba — APPROVED** [review]

**Pull request merged**